### PR TITLE
New API's and merged

### PR DESCRIPTION
**Pull request by radeonxray**
Added a bunch of new API's, not all totally optimsed.
Merged with William's latest work
Created a serperate "Queries"-fodler for alle queries. All Williams Queries have remained untouched in the original file, which is now placed in that folder.
Need to comment more on the different API's to specify whats going on.

### CODE REVIEW

File: src/controllers/mysql/queries/postAndCommentsQueries.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { PostObject } from '../../../types/post'
+import connection from '../connection';
+import { resolve } from 'url';
+
+
+export function selectAllPostAndComments_ByUser_name(username:string){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/postAndCommentsQueries.ts]: Consistant JS functions names, use camel case only (also better name?)
selectAllContentByUsername

File: src/controllers/mysql/queries/postAndCommentsQueries.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    'where user.username =?'
+    , [username],(error, results, fields) =>{
+      let post_and_comments_byuser_name = results;
+      resolve(post_and_comments_byuser_name)
+    })
+    })
+  }
+
+
+export function selectAllPostAndComments_ByUser_id(userid:number){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/postAndCommentsQueries.ts]: selectAllContentById

File: src/controllers/mysql/queries/postAndCommentsQueries.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    'join post ON post.`id` = comment.`fk_post` '+
+    'where user.id =?'
+    , [userid],(error, results, fields) =>{
+      let post_and_comments_byuser_id = results;
+      resolve(post_and_comments_byuser_id)
+    })
+    })
+  }
+  
+  export function selectAllPostAndComments_ByPost_id(postid:number){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/postAndCommentsQueries.ts]: selectALlContentByPostId

File: src/controllers/mysql/queries/postAndCommentsQueries.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    'join post ON post.`id` = comment.`fk_post` '+
+    'where post.id =?'
+    , [postid],(error, results, fields) =>{
+      let post_and_comments_bypost_id = results;
+      resolve(post_and_comments_bypost_id)
+    })
+    })
+  }
+  
+  export function selectAllPostAndComments_ByPost_title(postTitle:string){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/postAndCommentsQueries.ts]: selectAllContentByPostTitle

File: src/controllers/mysql/queries/postQueries.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import { PostObject } from '../../../types/post'
+import connection from '../connection';
+import { resolve } from 'url';
+
+
+
+
+export function selectAllUsers_And_Show_Posts_And_CommentAmount(){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/postQueries.ts]: selectSpecificUsersContentCount

File: src/controllers/mysql/queries/postQueries.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+     + ' join comment on comment.`fk_post` = post.id'
+     + ' GROUP BY post.id'
+      , (error, results, fields)=>{
+        let usersAndPost = results;
+        resolve(usersAndPost);
+      })
+    })
+  }
+  
+  export function selectAllUsers_And_Show_Posts(){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/postQueries.ts]: selectAllUsersAndPosts

File: src/controllers/mysql/queries/postQueries.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    return new Promise((resolve)=>{
+      connection.query('SELECT user.username AS user, post.title AS userpost, post.id as postId, post.url FROM user JOIN post ON user.id = post.fk_user'
+      , (error, results, fields)=>{
+        let usersAndPost = results;
+        resolve(usersAndPost);
+      })
+    })
+  }
+  
+  export function selectUser_ByID_Show_Posts(id: number){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/postQueries.ts]: selectPostsFromUserid

File: src/controllers/mysql/queries/postQueries.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    return new Promise((resolve) => {
+      connection.query('SELECT user.username AS user, post.title AS userpost, post.id as postId, post.url FROM user JOIN post ON user.id = post.fk_user WHERE user.id =?'
+      ,[id], (error, results, fields) => {
+        let userAndPost = results;
+        resolve(userAndPost);
+      })
+    })
+  }
+  
+  export function selectUser_ByName_Show_Posts(name: string){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/postQueries.ts]: selectUsernameFromPosts

File: src/controllers/mysql/queries/postQueries.ts [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+      ,[name], (error, results, fields) => {
+        let userAndPost = results;
+        resolve(userAndPost);
+      })
+    })
+  }
+
+
+
+export function selectPosts_ByTitle(title: string){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/postQueries.ts]: selectPostsFromTitle

File: src/controllers/mysql/queries/postQueries.ts [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    return new Promise((resolve) => {
+      connection.query('SELECT user.username AS user, post.title AS userpost, post.id as postId, post.url FROM user JOIN post ON user.id = post.fk_user WHERE post.title =?'
+      ,[title], (error, results, fields) => {
+        let PostByTitle = results;
+        resolve(PostByTitle);
+      })
+    })
+  }
+  
+  export function selectPosts_ById(postID: number){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/postQueries.ts]: selectPostsFromId

File: src/controllers/mysql/queries/userQueries.ts [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+  export function selectUserFromID(id: number){
+    return new Promise((resolve) =>{
+      connection.query('SELECT id,username,email,karma,role FROM user WHERE id =?', [id], (error, results, fields) => {
+        let user = results;
+        resolve(user);
+      })
+    })
+  }
+  
+  export function selectUsers_With_Above_Or_Equal_Karma_Points(karma: number){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/userQueries.ts]: selectUsersAboveKarma

File: src/controllers/mysql/queries/userQueries.ts [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+    return new Promise((resolve) =>{
+      connection.query('SELECT id,username,email,karma,role FROM user WHERE karma >=?', [karma], (error, results, fields) => {
+        let user = results;
+        resolve(user);
+      })
+    })
+  }
+  
+  export function selectUsers_With_Below_Or_Equal_Karma_Points(karma: number){
```
Comment (Viterkim) [inline src/controllers/mysql/queries/userQueries.ts]: selectUsersBelowKarma

File: src/controllers/mysql/queries/postAndCommentsQueries.ts (hunk 1 above)
Comment (radeonxray) [reply src/controllers/mysql/queries/postAndCommentsQueries.ts]: Totally understood. Gone through all myu functions and renamed to follow the convention and tried to make them easier to understand with shorter names and added comments above them

File: src/controllers/mysql/queries/postAndCommentsQueries.ts (hunk 2 above)
Comment (radeonxray) [reply src/controllers/mysql/queries/postAndCommentsQueries.ts]: Done

File: src/controllers/mysql/queries/postAndCommentsQueries.ts (hunk 4 above)
Comment (radeonxray) [reply src/controllers/mysql/queries/postAndCommentsQueries.ts]: Done

File: src/controllers/mysql/queries/postAndCommentsQueries.ts (hunk 3 above)
Comment (radeonxray) [reply src/controllers/mysql/queries/postAndCommentsQueries.ts]: Done

File: src/controllers/mysql/queries/postQueries.ts (hunk 5 above)
Comment (radeonxray) [reply src/controllers/mysql/queries/postQueries.ts]: Done

File: src/controllers/mysql/queries/postQueries.ts (hunk 8 above)
Comment (radeonxray) [reply src/controllers/mysql/queries/postQueries.ts]: Done

File: src/controllers/mysql/queries/postQueries.ts (hunk 10 above)
Comment (radeonxray) [reply src/controllers/mysql/queries/postQueries.ts]: Done

File: src/controllers/mysql/queries/postQueries.ts (hunk 9 above)
Comment (radeonxray) [reply src/controllers/mysql/queries/postQueries.ts]: Done

File: src/controllers/mysql/queries/userQueries.ts (hunk 11 above)
Comment (radeonxray) [reply src/controllers/mysql/queries/userQueries.ts]: Done

File: src/controllers/mysql/queries/userQueries.ts (hunk 12 above)
Comment (radeonxray) [reply src/controllers/mysql/queries/userQueries.ts]: Done

**Pull request merged**